airbyte-connector-builder-server/connector_builder/impl/default_api.py

```diff
@@ -27,7 +27,8 @@
 class DefaultApiImpl(DefaultApi):
     logger = logging.getLogger("airbyte.connector-builder")
 
-    def __init__(self, max_record_limit: int = 1000):
+    def __init__(self, adapter_cls: Type, max_record_limit: int = 1000):
```

Comment:
I think there are two problems with this type signature: 
1. `Type` is not very informative (is it the correct type? it seems imported from airbyte_cdk.models which doesn't seem right?) 
2. as a result it's not clear what methods can be called on that type and with what params (e.g: how does the reader know the `adapter_cls` can be called with a `manifest` argument?) 

Example of a signature which solves these might be: 

```
class CdkAdapter(abc):
  <..describe methods we expect this class to have..>

class DefaultApiImpl(..):
  def __init__(self, adapter_cls: Callable[[], CdkAdapter], ...
```

Reply:
Ah, good call @sherifnada - that was supposed to be lowercase `type`. I like your suggestion to use an interface to be clearer about the type and went ahead and made that change.